pkg/rlptools/wasm_utils.go

```diff
@@ -10,98 +10,100 @@ import (
-			Configurations: rlp.Spec.RateLimits[idx].Configurations,
-			Rules:          rules,
-		})
-	}
+type PatternOperator kuadrantv1beta2.WhenConditionOperator
 
-	if len(rlp.Spec.RateLimits) > 0 && len(flattenActions) == 0 {
-		// no configurations specified in the rlp,
-		// then apply the default configuration (action list) and default rules from the route
-		flattenActions = []GatewayAction{
-			{
-				Configurations: DefaultGatewayConfiguration(client.ObjectKeyFromObject(rlp)),
-				Rules:          HTTPRouteRulesToRLPRules(common.RulesFromHTTPRoute(route)),
-			},
-		}
-	}
+type PatternExpression struct {
+	// Selector of an attribute from the contextual properties provided by Envoy
+	// during request and connection processing
+	// https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/advanced/attributes
+	// They are named by a dot-separated path (e.g. request.path)
+	// Examples:
+	// "request.path" -> The path portion of the URL
+	Selector string `json:"selector"`
+
+	// The binary operator to be applied to the content fetched from context, for comparison with "value".
+	// Possible values are: "eq" (equal to), "neq" (not equal to), "incl" (includes; for arrays), "excl" (excludes; for arrays), "matches" (regex)
+	// TODO build comprehensive list of operators
+	Operator PatternOperator `json:"operator"`
 
-	return flattenActions
+	// The value of reference for the comparison with the content fetched from the context.
+	Value string `json:"value"`
 }
 
-func HTTPRouteRulesToRLPRules(httpRouteRules []common.HTTPRouteRule) []kuadrantv1beta1.Rule {
-	rlpRules := make([]kuadrantv1beta1.Rule, 0, len(httpRouteRules))
-	for idx := range httpRouteRules {
-		var tmp []string
-		rlpRules = append(rlpRules, kuadrantv1beta1.Rule{
-			// copy slice
-			Paths:   append(tmp, httpRouteRules[idx].Paths...),
-			Methods: append(tmp, httpRouteRules[idx].Methods...),
-			Hosts:   append(tmp, httpRouteRules[idx].Hosts...),
-		})
-	}
-	return rlpRules
+// Condition defines traffic matching rules
+type Condition struct {
+	// All of the expressions defined must match to match this condition
+	// +optional
+	AllOf []PatternExpression `json:"allOf,omitempty"`
+}
+
+// Rule defines one rate limit configuration. When conditions are met,
+// it uses `data` section to generate one RLS descriptor.
+type Rule struct {
+	// +optional
+	Conditions []Condition `json:"conditions,omitempty"`
+	// +optional
+	Data []DataItem `json:"data,omitempty"`
 }
 
 type RateLimitPolicy struct {
-	Name            string   `json:"name"`
-	RateLimitDomain string   `json:"rate_limit_domain"`
-	UpstreamCluster string   `json:"upstream_cluster"`
-	Hostnames       []string `json:"hostnames"`
+	Name      string   `json:"name"`
+	Domain    string   `json:"domain"`
+	Service   string   `json:"service"`
+	Hostnames []string `json:"hostnames"`
+
 	// +optional
-	GatewayActions []GatewayAction `json:"gateway_actions,omitempty"`
+	Rules []Rule `json:"rules,omitempty"`
```

Comment:
Are we also renaming `failure_mode_deny` and `rate_limit_policies` as in https://github.com/Kuadrant/wasm-shim/issues/35#issuecomment-1587044143?